### PR TITLE
Update DG user stories, glossary, NFRs, use cases

### LINKED ISSUE #17: Update the DG: user stories, glossary, NFRs, use cases
Update the DG based on project notes
- target profiles, value proposition and user stories
- use cases
- Non functional requirements
- glossary: define terms that are worth recording

### PR DESCRIPTION
**Pull request by guangxuan**
Closes #17

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -257,71 +257,92 @@ _{Explain here how the data archiving feature will be implemented}_
...
+
 
 *{More to be added}*
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `PropertyWhiz` and the **Actor** is the `user`, unless specified otherwise)
 
 **Use case: Delete a person**
```
Comment (lizchow) [inline docs/DeveloperGuide.md:294]:
```suggestion

```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -257,71 +257,92 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `H`  | user                                       | add a property                 | keep track of the property                                                 |
+| `H`  | user                                       | delete a property              | stop keeping track of the property                                         |
+| `H`  | user                                       | list all properties            | easily see all properties                                                  |
+| `H`  | user                                       | find property by name          | locate details of property without having to go through the entire list    |
+| `H`  | user                                       | update property tags           | provide details about a property                                           |
+| `H`  | user                                       | find property by tags          | easily see related properties with a given tag                             |
+| `H`  | user                                       | add, update, delete a buyer    | keep track of the buyer                                                    |
```
Comment (lizchow) [inline docs/DeveloperGuide.md:282]: Should we break `add, update, delete a buyer` into 3 separate user stories?

**Review by lizchow — COMMENTED** [review]
Here's some preliminary comments. Perhaps @sunjc826 can look through it too!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (sunjc826) [inline docs/DeveloperGuide.md:282]: To be consistent with properties, we can consider separating CUD buyer into separate user storeis as well.
```
| `H`  | user                                       | add   | keep track of the buyer                                                    |
| `H`  | user                                       | delete   | stop keeping track of the buyer                                                    |
| `H`  | user                                       | update   | keep up to date with the buyer's details                                                    |
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -257,71 +257,92 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `H`  | user                                       | update buyer tags              | provide details about a buyer                                              |
+| `H`  | user                                       | find buyer by tags             | easily see related buyers with a given tag                                 |
+
 
 *{More to be added}*
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `PropertyWhiz` and the **Actor** is the `user`, unless specified otherwise)
```
Comment (sunjc826) [inline docs/DeveloperGuide.md:292]: nitpick: `user` -> `User`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -257,71 +257,92 @@ _{Explain here how the data archiving feature will be implemented}_
...
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4. Should be able to function without internet access.
+5. Should be able to restart without loss of data.
 
 *{More to be added}*
 
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
-* **Private contact detail**: A contact detail that is not meant to be shared with others
```
Comment (sunjc826) [inline docs/DeveloperGuide.md:346]:
```markdown
- **Buyer**: A person who expresses interest in a range of properties
```

**Review by sunjc826 — CHANGES_REQUESTED** [review]
Looks good, some changes I suggested can be considered.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (guangxuan) [reply docs/DeveloperGuide.md:294]: Good catch.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (guangxuan) [reply docs/DeveloperGuide.md:282]: Sure thing.

Comment (guangxuan) [reply docs/DeveloperGuide.md:282]: Yea sure.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (guangxuan) [reply docs/DeveloperGuide.md:292]: mmm I prefer not to change this, it appears that the rest of the DG uses ```user``` currently.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (guangxuan) [reply docs/DeveloperGuide.md:346]: Yup. Great addition to the glossary.

**Review by sunjc826 — APPROVED** [review]
Looks good! Nothing to nitpick.

**Pull request merged**